src/workspace.ts

```diff
@@ -270,65 +294,55 @@ export class Workspace {
    */
   async run(
       workspaceRepos: WorkspaceRepo[],
-      fn: (repo: WorkspaceRepo) => Promise<void>):
-      Promise<BatchProcessResponse> {
+      fn: (repo: WorkspaceRepo) => Promise<void>,
+      concurrency = 1): Promise<BatchProcessResponse<WorkspaceRepo>> {
     this.assertInitialized();
-    const successRuns = [];
-    const failRuns = new Map<WorkspaceRepo, Error>();
-    for (const workspaceRepo of workspaceRepos) {
-      try {
-        await fn(workspaceRepo);
-        successRuns.push(workspaceRepo);
-      } catch (err) {
-        failRuns.set(workspaceRepo, err);
-      }
-    }
-    return [successRuns, failRuns];
+    return batchProcess(workspaceRepos, fn, {concurrency});
   }
 
   /**
    * Create a new branch on each repo.
    */
-  async startNewBranch(workspaceRepos: WorkspaceRepo[], newBranch: string) {
+  async startNewBranch(workspaceRepos: WorkspaceRepo[], newBranch: string):
+      Promise<BatchProcessResponse<WorkspaceRepo>> {
     this.assertInitialized();
-    await Promise.all(workspaceRepos.map((repo) => {
+    return batchProcess(workspaceRepos, (repo) => {
```

Comment:
yeah, based on this and my previous comment, it might even be more ergonomic to rate limit inside `GitRepo` and friends.